src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -3937,7 +3971,12 @@ class C
                 Await state.WaitForAsynchronousOperationsAsync()
 
                 Assert.Equal("#", state.GetLineFromCurrentCaretPosition().GetText())
-                Await state.AssertNoCompletionSession()
+                If completionImplementation = CompletionImplementation.Legacy Then
+                    Await state.AssertNoCompletionSession()
+                Else
+                    Await state.AssertCompletionSession() ' Agreed on changing the behavior
```

Comment:
what behavor change is there.  why do we have a change in behavior between the existing systme and the replacement.  Note: the existing system is not "legacy".  It is the oracle that the new system needs to match to ensure that muscle memory is not broken for people.

Reply:
Currently, we come back to the issues for the fourth time in the review.

in 
class C{
    $$
}
type '#'

In both cases, you'll be move to the left. The existing completion kills the completion session. The new one keeps the session. In the previous iteration (number 3) of considering this issue, both @cyrusnajmabadi and @sharwell agreed that the change is an improvement and suggested to correct the existing behavior to match it. However, it is hard to correct the existing behavior without affecting other scenario (which should not be affected). Also it is hard to match the existing behavior in the new implementation: editor does not provide us OnCaretPositionChanged.

Therefore, we (@jinujoseph, @amadeusw and @ivanbasov) agreed to keep the existing behavior and consider the new behavior as an improvement.

---
In reply to: [232508878](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508878) [](ancestors = 232508878)